internal/db/config.go

```diff
@@ -19,12 +19,16 @@ const (
 	updateEventBufferSize = 100
 )
 
+type dbOptions struct {
+	maxTxnRetries immutable.Option[int]
+}
+
 // Option is a funtion that sets a config value on the db.
-type Option func(*db)
+type Option func(*dbOptions)
 
 // WithMaxRetries sets the maximum number of retries per transaction.
 func WithMaxRetries(num int) Option {
-	return func(db *db) {
-		db.maxTxnRetries = immutable.Some(num)
+	return func(opts *dbOptions) {
+		opts.maxTxnRetries = immutable.Some(num)
```

Comment:
I considered adding KMS options as part of db options and noticed that it's not quite flexible, as it didn't allow post-processing of options, since it directly manipulates the values on `db` object.
So I changed it. But then I decided to move KMS elsewhere.
I can revert this change to options, but this style also seems to align with the way we do it in other places.